aiida_mlip/parsers/md_parser.py

```diff
@@ -0,0 +1,116 @@
+"""
+MD parser.
+"""
+
+from pathlib import Path
+
+import yaml
+
+from aiida.common import exceptions
+from aiida.engine import ExitCode
+from aiida.orm import Dict, SinglefileData
+from aiida.orm.nodes.process.process import ProcessNode
+from aiida.plugins import CalculationFactory
+
+from aiida_mlip.calculations.md import MD
+from aiida_mlip.helpers.converters import xyz_to_aiida_traj
+from aiida_mlip.parsers.base_parser import BaseParser
+
+MDCalculation = CalculationFactory("janus.md")
+
+
+class MDParser(BaseParser):
+    """
+    Parser class for parsing output of molecular dynamics simulation.
+
+    Inherits from SPParser.
+
+    Parameters
+    ----------
+    node : aiida.orm.nodes.process.process.ProcessNode
+        ProcessNode of calculation.
+
+    Methods
+    -------
+    parse(**kwargs: Any) -> int:
+        Parse outputs, store results in the database.
+
+    Returns
+    -------
+    int
+        An exit code.
+
+    Raises
+    ------
+    exceptions.ParsingError
+        If the ProcessNode being passed was not produced by a `MD`.
+    """
+
+    def __init__(self, node: ProcessNode):
+        """
+        Check that the ProcessNode being passed was produced by a `MD`.
+
+        Parameters
+        ----------
+        node : aiida.orm.nodes.process.process.ProcessNode
+            ProcessNode of calculation.
+        """
+        super().__init__(node)
+
+        if not issubclass(node.process_class, MDCalculation):
+            raise exceptions.ParsingError("Can only parse `MD` calculations")
+
+    def parse(self, **kwargs) -> ExitCode:
+        """
+        Parse outputs, store results in the database.
+
+        Parameters
+        ----------
+        **kwargs : Any
+            Any keyword arguments.
+
+        Returns
+        -------
+        int
+            An exit code.
+        """
+        # Call the parent parse method to handle common parsing logic
+        exit_code = super().parse(**kwargs)
+        if exit_code == ExitCode(0):
+
```

Comment:
This would probably be better as:
```suggestion
        if exit_code != ExitCode(0):
            return exit_code
```
And de-indent the rest of the function